cmd/monitorpartner.go

```diff
@@ -0,0 +1,171 @@
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path"
+	"regexp"
+
+	"github.com/VirusTotal/vt-cli/utils"
+	vt "github.com/VirusTotal/vt-go"
+	"github.com/cavaliercoder/grab"
+	"github.com/fatih/color"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+var monitorPartnerItemsCmdExample = `  vt monitorpartner list
+  vt monitorpartner list --filter "engine:<EngineName>" --include sha256,first_detection_date`
+
+// NewMonitorPartnerHashesListCmd returns a list or monitor_partner according to a filter.
+func NewMonitorPartnerHashesListCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:     "list",
+		Short:   "List available monitor partner hashes",
+		Example: monitorItemsCmdExample,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			p, err := NewPrinter(cmd)
+			if err != nil {
+				return err
+			}
+			return p.PrintCollection(vt.URL("monitor_partner/hashes"))
+		},
+	}
+
+	addIncludeExcludeFlags(cmd.Flags())
+	addFilterFlag(cmd.Flags())
+	addLimitFlag(cmd.Flags())
+	addCursorFlag(cmd.Flags())
+
+	return cmd
+}
+
+// Monitor Partner downloader, it implements the Doer interface. Retrieves the item
+// path to know the destination filename and downloads and save each individual
+// file using fileDownloader.DownloadFile
+type monitorPartnerDownloader struct {
+	fileDownloader
+}
+
+func (d *monitorPartnerDownloader) Do(file interface{}, ds *utils.DoerState) string {
+	var hash string
+	if f, isObject := file.(*vt.Object); isObject {
+		hash = f.ID()
+	} else {
+		hash = file.(string)
+	}
+
+	ds.Progress = fmt.Sprintf("%s %4.1f%%", hash, 0.0)
+
+	// Get download URL
+	var downloadURL string
+	_, err := d.client.GetData(vt.URL("monitor_partner/files/%s/download_url", hash), &downloadURL)
+
+	if err == nil {
+		dstPath := path.Join(viper.GetString("output"), hash)
+		err = d.DownloadFile(downloadURL, dstPath, func(resp *grab.Response) {
+			progress := 100 * resp.Progress()
+			if progress < 100 {
+				ds.Progress = fmt.Sprintf("%s %4.1f%% %6.1f KBi/s",
+					hash, progress, resp.BytesPerSecond()/1024)
+			}
+		})
+	}
+
+	msg := color.GreenString("ok")
+	if err != nil {
+		if apiErr, ok := err.(vt.Error); ok && apiErr.Code == "NotFoundError" {
+			msg = color.RedString("not found")
+		} else {
+			msg = color.RedString(err.Error())
+		}
+	}
+
+	return fmt.Sprintf("%s [%s]", hash, msg)
+}
+
+var monitorPartnerHashDownloadCmdHelp = `Download files from your partner account.
+
+This command download files in your monitor partner account using their sha256.`
```

Comment:
"download files in your monitor partner account" -> "download files **from** your monitor partner account"

Reply:
corrected